### DISCUSSION TITLE
Dialog Box onClose event

### DISCUSSION THREAD

**Discussion by willwainwright** [microsoft/Microsoft-Fabric-workload-development-sample#31] [Help] [answered]
Similar to another question regarding page navigation, we have an issue where a react component does not rerender when a dialog is opened then closed.

We use the workloadClient.dialog.open to open a IFrame dialog and displays a component. However if a user closes a dialog (either from the close or by clicking away), if they re-open the dialog the previous state is shown. Is there a function to detect when a dialog is closed to clean the state?

**Comment by ramizsha** [comment, marked as answer]
Hi @willwainwright,
workloadClient.dialog.open returns a promise that resolves after the dialog is closed, you can use this to clean the state.
